packages/toolpad-core/src/Crud/Show.tsx

```diff
@@ -0,0 +1,262 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Alert from '@mui/material/Alert';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CircularProgress from '@mui/material/CircularProgress';
+import Divider from '@mui/material/Divider';
+import Grid from '@mui/material/Grid2';
+import Paper from '@mui/material/Paper';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import EditIcon from '@mui/icons-material/Edit';
+import DeleteIcon from '@mui/icons-material/Delete';
+import invariant from 'invariant';
+import dayjs from 'dayjs';
+import { useDialogs } from '../useDialogs';
+import { useNotifications } from '../useNotifications';
+import { DataField, DataModel, DataModelId, DataSource } from './shared';
+import { CrudContext } from '../shared/context';
+
+export interface ShowProps<D extends DataModel> {
+  id: DataModelId;
+  /**
+   * Server-side data source.
+   */
+  dataSource?: DataSource<D> & Required<Pick<DataSource<D>, 'getOne'>>;
+  /**
+   * Callback fired when the "Edit" button is clicked.
+   */
+  onEditClick?: (id: DataModelId) => void;
+  /**
+   * Callback fired when the item is successfully deleted.
+   */
+  onDelete?: (id: DataModelId) => void;
+}
+/**
+ *
+ * Demos:
+ *
+ * - [Crud](https://mui.com/toolpad/core/react-crud/)
+ *
+ * API:
+ *
+ * - [Show API](https://mui.com/toolpad/core/api/show)
+ */
+function Show<D extends DataModel>(props: ShowProps<D>) {
+  const { id, onEditClick, onDelete } = props;
+
+  const crudContext = React.useContext(CrudContext);
+  const dataSource = (props.dataSource ?? crudContext.dataSource) as Exclude<
+    typeof props.dataSource,
+    undefined
+  >;
+
+  invariant(dataSource, 'No data source found.');
+
+  const { fields, ...methods } = dataSource;
+  const { getOne, deleteOne } = methods;
+
+  const dialogs = useDialogs();
+  const notifications = useNotifications();
+
+  const [data, setData] = React.useState<D | null>(null);
```

Comment:
Not using any data fetching library (such as `react-query`) for now, so no caching of requests, for example.
This simple implementation could be a placeholder for using the MUI X data providers when they're ready.

Reply:
Update: I've added caching with the `dataSourceCache` prop based on the approach from https://mui.com/x/react-data-grid/server-side-data/?srsltid=AfmBOooZZzlgdHeSqhZ6ZM7J9SKLqXZeo6W07kidqbp80tqql3xIeSHc#data-caching.

It's documented in the `Data caching` section of the component docs.

Reply:
Makes sense, if someone wants to use a custom cache (as documented in the link above) they could pass in a custom Datagrid in the List component's `slot`, right?

Reply:
I didn't document using a custom cache but it would be same as the approach in that page. I'm recommending that users pass an instance like `new DataSourceCache()` to the `dataSourceCache` prop but they could use a different class with the same methods if they want to.

In the end we would just use the same exact cache feature as MUI X once they finish the server-side data and make it free/public, so this is kind of a placeholder.